### PR TITLE
refactor: move oobv2 'From' field up from attachment

### PR DESCRIPTION
**Pull request by Baha-sk**
As per the DIDComm V2 spec, the From field should be in the oobv2 struct directly, not in the attachment.

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: pkg/didcomm/protocol/outofbandv2/service.go [hunk 1]
```diff
@@ -242,76 +246,46 @@ func (s *Service) AcceptInvitation(i *Invitation) (string, error) { // nolint: g
...
+		return "", fmt.Errorf("oob/2.0 failed to create destination: %w", err)
+	}
 
-		if err := s.connectionRecorder.SaveConnectionRecord(connRecord); err != nil {
-			return "", err
-		}
+	connRecord := &connection.Record{
+		ConnectionID:      uuid.New().String(),
+		ParentThreadID:    i.ID,
+		State:             "null",
```
Comment (Moopli) [inline pkg/didcomm/protocol/outofbandv2/service.go:267]: I was wondering, why do you set state "null"? This confuses the `connection.Recorder`, which only saves the record with certain metadata (like, with all the tags needed for lookups elsewhere in the code) when the state is 'completed'.

Comment (Baha-sk) [reply pkg/didcomm/protocol/outofbandv2/service.go]: updated to `connection.StateNameCompleted`

Comment (Moopli) [reply pkg/didcomm/protocol/outofbandv2/service.go]: (resolved now)

**Review by Moopli — APPROVED** [review]
lgtm :+1:

File: pkg/didcomm/protocol/outofbandv2/service.go [hunk 2]
```diff
@@ -242,76 +246,46 @@ func (s *Service) AcceptInvitation(i *Invitation) (string, error) { // nolint: g
...
+	connRecord := &connection.Record{
+		ConnectionID:      uuid.New().String(),
+		ParentThreadID:    i.ID,
+		State:             connection.StateNameCompleted,
+		InvitationID:      i.ID,
+		ServiceEndPoint:   destination.ServiceEndpoint,
+		RecipientKeys:     destination.RecipientKeys,
+		RoutingKeys:       destination.RoutingKeys,
+		TheirLabel:        i.Label,
+		TheirDID:          i.From,
```
Comment (Moopli) [inline pkg/didcomm/protocol/outofbandv2/service.go:273]: ah wait, don't we also need to set MyDID in the record?

Comment (Baha-sk) [reply pkg/didcomm/protocol/outofbandv2/service.go:273]: technically it's the same as `InvitationDID` further below... but i've added MyDID for the sake of completeness.

Comment (Moopli) [reply pkg/didcomm/protocol/outofbandv2/service.go:273]: :+1: yeah it's a bit of cruft from the record being used in different ways in didexchange vs elsewhere
but `MyDID` is needed in a lot of places, while `InvitationDID` is mostly ignored.

**Review by Moopli — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Pull request merged**